sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/controller/container/ThroughputContainerController.java

```diff
@@ -0,0 +1,321 @@
+
+        return Mono.just(request)
+            .flatMap(request1 -> {
+                if (request1.getThroughputControlGroupName() == null) {
+                    return Mono.just(new Utils.ValueHolder<>(this.defaultGroupController));
+                }
+                else {
+                    // If customer used a group not defined, we will fall back to the default group controller
+                    return Mono.justOrEmpty(this.groupControllers.get(request1.getThroughputControlGroupName()))
+                        .defaultIfEmpty(this.defaultGroupController)
+                        .map(Utils.ValueHolder::new);
+                }
+            })
+            .flatMap(groupController -> {
+                if (groupController.v != null) {
+                    return groupController.v.processRequest(request, originalRequestMono);
+                }
+
+                return originalRequestMono;
+            });
+    }
+
+    public String getTargetContainerRid() {
+        return this.targetContainerRid;
+    }
+
+    @Override
+    public boolean canHandleRequest(RxDocumentServiceRequest request) {
+        checkNotNull(request, "Request can not be null");
+
+        return StringUtils.equals(this.targetContainerRid, request.requestContext.resolvedCollectionRid);
+    }
+
+    private Mono<ThroughputContainerController> createAndInitializeGroupControllers() {
+        return Flux.fromIterable(this.groups)
+            .flatMap(group -> {
+                ThroughputGroupControllerBase groupController =
+                    this.groupControllers.computeIfAbsent(
+                        group.getGroupName(),
+                        groupName -> ThroughputGroupControllerFactory.createController(
+                                        this.connectionMode,
+                                        this.globalEndpointManager,
+                                        group,
+                                        this.maxContainerThroughput.get(),
+                                        this.partitionKeyRangeCache,
+                                        this.targetContainerRid));
+                return Mono.just(groupController);
+            })
+            .flatMap(groupController -> groupController.init())
+            .then(Mono.just(this));
+    }
+
+    private Flux<Void> refreshContainerMaxThroughputTask(CancellationToken cancellationToken) {
+        checkNotNull(cancellationToken, "Cancellation token can not be null");
+
+        if (this.throughputResolveLevel == ThroughputResolveLevel.NONE) {
+            return Flux.empty();
+        }
+
+        return Mono.delay(DEFAULT_THROUGHPUT_REFRESH_INTERVAL)
+            .flatMap(t -> this.resolveContainerMaxThroughput())
+            .flatMapIterable(controller -> this.groupControllers.values())
+            .doOnNext(groupController -> groupController.onContainerMaxThroughputRefresh(this.maxContainerThroughput.get()))
+            .onErrorResume(throwable -> {
```

Comment:
This is the correct usage of `onErrorResume()` :) 
As on error here, you are resuming your reactive stream with an empty Mono.

Reply:
yup yup, that is the intent here